### DISCUSSION TITLE
How to read "perf_graph"

### DISCUSSION THREAD

**Discussion by styyokuda** [idaholab/moose#25902] [Q&A General]
All,

I am trying to measure the CPU time for a run. I believe the “perf_graph” object does.
I got the attached table for my run (please see the attached file).
Because I am not sure how to read this table, please help.
Form the table, “Total (s) = 11469.209.” Also, I used 12 processors. Therefore, 11469.209 x 12 = 137,630.508 seconds = 38.23 hours.
Please let me know if I am right.
S. Thomas
[perf_graph.pdf](https://github.com/idaholab/moose/files/13210375/perf_graph.pdf)

**Comment by GiudGiud** [comment]
Hello

Assuming you only used 1 thread per process, it would be 38 CPU hours.
Did the calculation take roughly 3 hours of real time?

80% of the run time is not accounted for though, it s not kernels for residuals, it s not for the Jacobian
Something is slowing this down a lot and we dont know what from the perf graph.
You could use a real profiler:
https://mooseframework.inl.gov/application_development/profiling.html

Guillaume
